ring-core/src/ring/util/response.clj

```diff
@@ -195,20 +219,29 @@
              dir-entry  (.getEntry jar-file (add-ending-slash entry-name))]
          (and dir-entry (.isDirectory dir-entry)))))
 
+(defmethod resource-info :jar
+    [^java.net.URL url]
+    (let [conn (.openConnection url)
+          dir? (jar-directory? conn)]
+      {:content        (if-not dir? (.getInputStream conn))
```

Comment:
I like that. My original thinking was that `resource-info` would be a useful public function, and should therefore give you data on directories as well, but I no longer think it's something ring should provide, and should just be an internal knob that has to be exposed for `resource-response` to function properly. So it should just do what it needs to do to meet that need, and no more.